### PR TITLE
Feat: token list add sequence No. column

### PR DESCRIPTION
**Pull request by JeffreyMa597**
What problem does this PR solve?
------
Add sequence No. column in the all pages of `token lists` and `contract list`.

image

Ref: https://github.com/Magickbase/godwoken_explorer/issues/481

Check List
------
As description.

### Test
e2e Test
### Task
none

### CODE REVIEW

**Comment by Keith-CY** [comment]
> @JeffreyMa597 @Keith-CY
> 
> 1. Is the column name the same as in the screenshot, and # is replaced by Rank?
>    [v1.testnet.gwscan.com/token/375?tab=holders&id=375&page=1](https://v1.testnet.gwscan.com/token/375?tab=holders&id=375&page=1)
> 
> image


Sure, `Rank` is better than `#` because `Rank` implies it's about the order but not the id

> 2. If the page displays 100 pieces of data, then 30 pieces are one page. Whether each page is 1-30 or displayed in positive order (for example, the first page (1-30), the second page (31-60), etc. Refer to [v1.testnet.gwscan.com/token/375?tab=holders&id=375&page=3](https://v1.testnet.gwscan.com/token/375?tab=holders&id=375&page=3)), it is the same if it is changed to 15 or 50 pages.
>    [godwoken-explorer-fqz6lqg2s-magickbase.vercel.app/nft-collections](https://godwoken-explorer-fqz6lqg2s-magickbase.vercel.app/nft-collections)
>    [godwoken-explorer-fqz6lqg2s-magickbase.vercel.app/multi-token-collections?after=g3QAAAABZAACaWRiAABZJA%3D%3D](https://godwoken-explorer-fqz6lqg2s-magickbase.vercel.app/multi-token-collections?after=g3QAAAABZAACaWRiAABZJA%3D%3D)
>    [godwoken-explorer-fqz6lqg2s-magickbase.vercel.app/contracts?page=2](https://godwoken-explorer-fqz6lqg2s-magickbase.vercel.app/contracts?page=2)

The rank range should be `page_size * (page_no - 1) + 1 to page_size * page_no`, so the `page 2` with `page_size = 30` should include `rank 31 - 60` and `page 2` with `page_size = 15` should include `rank 16 - 30`

But when it was on `page 2 with page_size 30` from `page 1`, namely the page is located with `after` cursor, and change the `page_size` to `50`, the rank should be `rank 31 - 80`.

If it was on `page 2 with page_size 30` from `page 3`, namely the page is located with `before` cursor, and change the `page_size` to `50`, the rank should be `rank 11 - 60`

File: pages/contracts.tsx [hunk 1]
```diff
@@ -111,6 +111,7 @@ const ContractList = () => {
             <Table>
               <thead style={{ textTransform: 'capitalize', fontSize: isMobile ? 12 : 14 }}>
                 <tr>
+                  <th style={{ width: '1rem' }}>{t(`#`)}</th>
```
Comment (Keith-CY) [inline pages/contracts.tsx:114]: Why fix the width of this column, and how would it be if `number` is great that width of `number` is wider than `1rem`

File: pages/contracts.tsx [hunk 2]
```diff
@@ -132,8 +133,9 @@ const ContractList = () => {
                     </tr>
                   ))
                 ) : data.meta.totalPage ? (
-                  data.contracts.map(c => (
+                  data.contracts.map((c, idx) => (
                     <tr key={c.id} title={c.address}>
+                      <td>{idx + 1}</td>
```
Comment (Keith-CY) [inline pages/contracts.tsx:138]: The index would repeat if `page number` is not considered

Comment (JeffreyMa597) [reply pages/contracts.tsx:138]: ok. I got it.

File: pages/contracts.tsx (hunk 1 above)
Comment (JeffreyMa597) [reply pages/contracts.tsx:114]: If we didn't give a width, this width may not be uniform for each page.
I'll remove it, first.

Comment (Keith-CY) [reply pages/contracts.tsx:114]: A `min-width` could help various pages to have the same width while keep it flexible for large number which requires more width

File: pages/contracts.tsx (hunk 2 above)
Comment (JeffreyMa597) [reply pages/contracts.tsx:138]: @Keith-CY Hold a minute. we are now using graphql requests to get data, and we paginating by the cursor. So, how can we get the current page number in the frontend?

Comment (Keith-CY) [reply pages/contracts.tsx:138]: Good question, it seems extremely hard to set a rank, although ranked tokens/contracts by time are easy, sorted by balance or holders, etc. are difficult, any idea or advice from @Naupio

**Comment by JeffreyMa597** [comment]
@Keith-CY hi~ Finally, we still mark the list by `index`, is that right?

**Pull request closed without merging**